src/vmod_cookie.c

```diff
@@ -248,21 +252,24 @@ vmod_get_re(VRT_CTX, struct vmod_priv *priv, struct vmod_priv *priv_call, VCL_ST
 		return(NULL);
 
 	if (priv_call->priv == NULL) {
+		AZ(pthread_mutex_lock(&mtx));
 		vre = compile_re(ctx, expression);
 		if (!vre)
 			return(NULL);   // Not much else to do, error already logged.
 
 		priv_call->priv = vre;
 		priv_call->free = free;
+		AZ(pthread_mutex_unlock(&mtx));
 	}
 
 	VTAILQ_FOREACH(current, &vcp->cookielist, list) {
 		CHECK_OBJ_NOTNULL(current, VMOD_COOKIE_ENTRY_MAGIC);
-		VSLb(ctx->vsl, SLT_VCL_Log, "cookie: checking %s", current->name);
+
+		// VSLb(ctx->vsl, SLT_VCL_Log, "cookie: checking %s", current->name);
 		i = VRE_exec(vre, current->name, strlen(current->name), 0, 0, ovector,
 				     VRE_MAX_GROUPS, NULL);
 		if (i >= 0) {
-			VSLb(ctx->vsl, SLT_VCL_Log, "cookie: %s is a match for regex '%s'", current->name, expression);
+			// VSLb(ctx->vsl, SLT_VCL_Log, "cookie: %s is a match for regex '%s'", current->name, expression);
```

Comment:
if you really want to keep your debug logging in, maybe use a macro which is only enabled by a define?